modules/core/src/testFixtures/java/org/apache/ignite/internal/testframework/log4j2/TestLogChecker.java

```diff
@@ -0,0 +1,324 @@
+import org.apache.logging.log4j.core.Appender;
+import org.apache.logging.log4j.core.LogEvent;
+import org.apache.logging.log4j.core.Logger;
+import org.apache.logging.log4j.core.appender.AbstractAppender;
+import org.apache.logging.log4j.core.config.Configuration;
+import org.apache.logging.log4j.core.config.LoggerConfig;
+import org.apache.logging.log4j.core.config.Property;
+
+/**
+ * Test log checker.
+ */
+public class TestLogChecker {
+    /** Logger name. */
+    private final String loggerName;
+
+    /** List of handlers. */
+    private final List<Handler> handlers = new ArrayList<>(1);
+
+    /** Lock that is used to synchronize the collection of handlers. */
+    private final ReadWriteLock lock = new ReentrantReadWriteLock();
+
+    /** Test log appender that is used to check log events/messages. */
+    private final TestLogAppender appender;
+
+    /** Logger configuration. */
+    private Configuration config;
+
+    /**
+     * Creates a new instance of {@link TestLogChecker} for the given {@code loggerName}.
+     *
+     * @return New instance of {@link TestLogChecker}.
+     **/
+    public static TestLogChecker create(String loggerName) {
+        return create(loggerName, false);
+    }
+
+    /**
+     * Creates a new instance of {@link TestLogChecker} for the given {@code loggerName}.
+     *
+     * @param started Whether the checker should be started.
+     * @return New instance of {@link TestLogChecker}.
+     **/
+    public static TestLogChecker create(String loggerName, boolean started) {
+        TestLogChecker checker = new TestLogChecker(loggerName);
+
+        if (started) {
+            checker.start();
+        }
+
+        return checker;
+    }
+
+    /**
+     * Creates a new instance of {@link TestLogChecker} for the given {@code clazz}.
+     *
+     * @return New instance of {@link TestLogChecker}.
+     **/
+    public static TestLogChecker create(Class<?> clazz) {
+        return create(clazz, false);
+    }
+
+    /**
+     * Creates a new instance of {@link TestLogChecker} for the given {@code clazz}.
+     *
+     * @param started Whether the checker should be started.
+     * @return New instance of {@link TestLogChecker}.
+     **/
+    public static TestLogChecker create(Class<?> clazz, boolean started) {
+        return create(clazz.getName(), started);
+    }
+
+    /**
+     * Creates a new instance of {@link TestLogChecker}.
+     *
+     * @param loggerName Logger name.
+     */
+    public TestLogChecker(String loggerName) {
+        Objects.requireNonNull(loggerName);
+
+        this.loggerName = loggerName;
+        this.appender = new TestLogAppender(loggerName);
+    }
+
+    /**
+     * Creates a new instance of {@link TestLogChecker} with the given {@code predicate}.
+     *
+     * @param loggerName Logger name.
+     * @param predicate Predicate to check log messages.
+     */
+    public TestLogChecker(String loggerName, Predicate<String> predicate) {
```

Comment:
> What happens if the predicate returns true or false?
updated javadoc in the following way:
```
    /**
     * Creates a new instance of {@link TestLogChecker} with the given {@code predicate}.
     * In order to check that the required log event was logged, use {@link #isMatched()} method.
     *
     * @param loggerName Logger name.
     * @param predicate Predicate to check log messages.
     */
```

> Also, in some cases it is required to verify other message attributes, not only its text
It makes sense to me. Predicate should accept `LogEvent` instead of `String`